src/lib/network_pool/snark_pool.ml

```diff
@@ -106,27 +113,37 @@ struct
     module T = struct
       (* TODO : Version this type *)
       type serializable =
-        Ledger_proof.Stable.V1.t One_or_two.Stable.V1.t
-        Priced_proof.Stable.V1.t
-        Statement_table.t
+        { all:
+            Ledger_proof.Stable.V1.t One_or_two.Stable.V1.t
+            Priced_proof.Stable.V1.t
+            Statement_table.t
+              (** Every SNARK in the pool *)
+        ; rebroadcastable:
+            ( Ledger_proof.Stable.V1.t One_or_two.Stable.V1.t
+              Priced_proof.Stable.V1.t
+            * Time.Stable.With_utc_sexp.V2.t )
+            Statement_table.t
+              (** Rebroadcastable SNARKs generated on this machine, along with
+                  when they were first added. *)
```

Comment:
Could we put a debug assert somewhere that this remains a subset of `all` or would that be too expensive?

Reply:
I added an assert in the snark pool when it drops stuff and one in the unit tests for transaction pool. The latter caught a bug. In the snark pool the complexity of the check is linear in the number of locally generated snarks, in the transaction pool it's approximately n log m.